src/topology/config/watcher.rs

```diff
@@ -0,0 +1,113 @@
+use super::Config;
+use futures::{Async, Future, Stream};
+use notify::{watcher, DebouncedEvent, RecursiveMode, Watcher};
+use std::{
+    path::Path,
+    sync::mpsc::{channel, TryRecvError},
+    time::{Duration, Instant},
+};
+use tokio::timer::Delay;
+
+// Per notify own documentation, it's advised to have delay of more than 30 sec,
+// so to avoid receiving repetitions of previous events on macOS.
+// Larger delays hurt responsivity, but that's fine as this is primarily designed
+// for cloud usage which isn't exactly super responsive.
+const CONFIG_WATCH_DELAY: std::time::Duration = std::time::Duration::from_secs(31);
+
+/// Returns when config should be reloaded.
+/// Never errors, and never ends.
+pub fn config_watcher(
```

Comment:
can we just combine this with the fn below, since its private?

Reply:
I wouldn't want to expose delay variable to the outside, and the test needs to be able to specify the delay so it could be done faster. That's why there are two of them.

Reply:
I think that is fine, if its a const, the main can just use the const from this file.

Reply:
Ok